# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: testing/web-platform/mozilla/tests/editor/delete-space-after-double-click-selection.html
# Commit: 6fab61424d92
# Full Hash: 6fab61424d92dfeadd6732ccf6b125ff25f797b4
# Author: Jan-Niklas Jaeschke <jjaschke@mozilla.com>
# Date: 2022-11-12 09:47:29
# Description:
#   Bug 1800043: Fixed condition to determine end of selected string. r=masayuki
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D161881
# ==============================================================================

diff -r 0926249d2745 -r 6fab61424d92 testing/web-platform/mozilla/tests/editor/delete-space-after-double-click-selection.html
--- a/testing/web-platform/mozilla/tests/editor/delete-space-after-double-click-selection.html	Fri Nov 11 12:55:28 2022 +0000
+++ b/testing/web-platform/mozilla/tests/editor/delete-space-after-double-click-selection.html	Fri Nov 11 13:04:54 2022 +0000
@@ -105,7 +105,17 @@
         }
         return test_driver.send_keys(editableElement, deleteKey);
       };
-
+      if (editableElement.tagName.toLowerCase() == "div") {
+        promise_test(async t => {
+          setValue("<p>abc def<span></span></p>");
+          await doubleClickAndDelete("abc de", "def");
+          await waitForRender();
+          assert_equals(
+            getValue(),
+            "<p>abc</p>",
+            "The <span> at the end of the string must be removed, as well as the whitespace in between words.");
+        }, `${editableElement.tagName}: An empty span at the end of the selection should be considered end of selection!`);
+      }
       promise_test(async t => {
         setValue("one two");
         await doubleClickAndDelete("on", "one");
